keychain_p.h

```diff
@@ -15,7 +15,7 @@
 #include <QSettings>
 #include <QQueue>
 
-#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID)
+#if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN) && !defined(Q_OS_ANDROID) && !defined(Q_OS_HAIKU)
```

Comment:
Maybe worth doing a  `#define KEYCHAIN_XDG`  or something like that (maybe KEYCHAIN_DBUS).  Then we can  use that ifdef instead of repeating the same condition all over the place

Reply:
Yes, I didn't suggest it because the line didn't wrap yet but it's cleaner.
We can even have CMake do it.

Reply:
what about !KEYCHAIN_HAVE_NATIVE ?

Reply:
In a way, secret service is the native way,  so i guess that name is not good.   XDG is the name used for the typicall desktop environment on X11.

Reply:
Technically XDG is a set of standards from freedesktop.org , most of which written with X11 in mind (wonder what that will give with Wayland:D), but some of those could also possibly be supported in Haiku.
So I'm not sure it really fits.